File: AtCoder/ACP/BootCamp4b/Hard100_2/python/D_Card_Eater.py
from collections import Counter
N = int(input())
C = Counter(list(map(int, input().split())))

cnt = 0
for v in C.values():
    if v % 2 == 0:
        cnt += 1
res = len(C) - int(cnt % 2 != 0)
print(res)

File: AtCoder/ACP/BootCamp4b/Hard100_2/cpp/D_Card_Eater.cpp
#include <bits/stdc++.h>
#define MAX_N 100'000
using namespace std;
typedef long long ll;

ll N;
ll A[MAX_N];

void solve() {
  map<ll, ll> C;
  for (ll i = 0; i < N; ++i) {
    ++C[A[i]];
  }
  ll cnt = 0;
  for (auto p : C) {
    if (p.second % 2 == 0) ++cnt;
  }
  ll res = C.size() - (cnt % 2 != 0);
  printf("%lld\n", res);
}

int main() {
  cin >> N;
  for (ll i = 0; i < N; ++i) {
    cin >> A[i];
  }
  solve();
}